scipy/signal/_upfirdn_apply.pyx

```diff
@@ -65,6 +65,56 @@ def _output_len(Py_ssize_t len_h,
     return need
 
 
+@cython.cdivision(True)  # faster modulo
+@cython.boundscheck(False)  # designed to stay within bounds
+@cython.wraparound(False)  # we don't use negative indexing
+def _rat(x, tol=None):
```

Comment:
Can you comment on why this approach, passing through a float, is better than dividing out the gcd, and possibly restricting the denominator to keep the up/down values reasonable?  I assume that what this does, but I haven't worked through it myself other than to try the code out a little.